index.bs

```diff
@@ -7878,6 +8748,77 @@ partial dictionary MLOpSupportLimits {
   </details>
 </div>
 
+### tile ### {#api-mlgraphbuilder-tile}
+Repeat a tensor the given number of times along each dimension.
+
+<script type=idl>
+partial interface MLGraphBuilder {
+  MLOperand tile(MLOperand input,
+                 sequence<unsigned long> repetitions,
+                 optional MLOperatorOptions options = {});
+};
+
+partial dictionary MLOpSupportLimits {
+  MLSingleInputSupportLimits tile;
+};
+</script>
+
+<div dfn-for="MLGraphBuilder/tile(input, repetitions, options)" dfn-type=argument>
+    **Arguments:**
+        - <dfn>input</dfn>: an {{MLOperand}}. The input N-D tensor.
+        - <dfn>repetitions</dfn>: A count per dimension of how many times to repeat that dimension. The |repetitions| [=list/size=] must match the |input|'s [=MLOperand/rank=], using 1's for any axis that should retain the same size.
+        - <dfn>options</dfn>: an optional {{MLOperatorOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The reversed N-D tensor.
+</div>
+
+<table id=constraints-transpose class='data' link-for="MLGraphBuilder/tile(input, options)">
+  <caption>Constraints for {{MLGraphBuilder/tile()}}</caption>
+  <thead>
+    <tr>
+      <th>operand</th>
+      <th>[=/allowed data types=]</th>
+      <th>[=/allowed ranks=]</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>{{input}}</td>
+    <td>[=/any data type|any=]</td>
+    <td>[=/any rank|N=]</td>
+  </tr>
+  <tr>
+    <td>*output*</td>
+    <td>[=/same type as|same as=] {{input}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+</table>
+
+{{MLOpSupportLimits}} has the following members for {{MLGraphBuilder/tile()}}:
+<dl dfn-type=dict-member dfn-for=MLOpSupportLimits>
+    : <dfn>tile</dfn>
+    :: Support limits for operator {{MLGraphBuilder/tile()}}.
+</dl>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>tile(|input|, |repetitions|, |options|)</dfn> method steps are:
+  </summary>
+    1. If [=this=] [=MLGraphBuilder/can not build=], then [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+    1. If [=MLGraphBuilder/validating operand=] with [=this=] and |input| returns false, then [=exception/throw=] a {{TypeError}}.
+    1. If |repetitions|'s [=list/size=] is not equal to |input|'s [=MLOperand/rank=], then [=exception/throw=] a {{TypeError}}.
+    1. If |repetitions|'s values contain 0's, then [=exception/throw=] a {{TypeError}}.
+
+        Issue(391): If 0-size dimensions are allowed, revise these steps.
+
+    1. *Make graph connections:*
+        1. Let |output| be the result of [=copying an MLOperand=] given |input|.
```

Comment:
The output size should be calculated by input size * repetition, as Chromium prototype does: https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/public/cpp/graph_validation_utils.cc;l=2324

Reply:
This issue seems to still exist in the latest commit?

Reply:
Sorry, fixed.